consensus/enclave/impl/src/lib.rs

```diff
@@ -813,9 +817,23 @@ fn mint_output<T: Digestible>(
     };
 
     // Create a single TxOut
-    let output = TxOut::new(amount, recipient, &tx_private_key, Default::default())
+    let mut output = TxOut::new(amount, recipient, &tx_private_key, Default::default())
         .map_err(|e| Error::FormBlock(format!("AmountError: {:?}", e)))?;
 
+    // The output must conform to block version rules
+    if !block_version.e_memo_feature_is_supported() {
+        output.e_memo = None;
+    }
+
+    if !block_version.masked_token_id_feature_is_supported() {
+        output.masked_amount.masked_token_id.clear();
+        if amount.token_id != 0 {
+            return Err(Error::FormBlock(
+                "Cannot mint outputs for non-MOB tokens until they are supported".to_string(),
+            ));
+        }
+    }
+
```

Comment:
Would it make sense to validate the txout being generated here just to confirm that we never return an invalid txout?

Reply:
What sort of validation are you thinking about? I believe the `if` statements above provide a guarantee that the `TxOut` is valid with respect to the block version used.

Reply:
I see, you were probably referring to `validate_tx_out`. It might be worth to add that as a sanity, since I could see a world in which we update our validation rules inside `validate_tx_out` but forget to update this function.

Reply:
i think the usual way of doing things is, we have unit tests to check things like that, and in prod we don't run the tests every time the function is called, because then we don't have a performance penalty in prod